// @trycourier/courier/resources/profiles/profiles.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as ListsAPI from "./lists.mjs";
import { Lists, } from "./lists.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Profiles extends APIResource {
    constructor() {
        super(...arguments);
        this.lists = new ListsAPI.Lists(this._client);
    }
    /**
     * Merge the supplied values with an existing profile or create a new profile if
     * one doesn't already exist.
     */
    create(userID, body, options) {
        return this._client.post(path `/profiles/${userID}`, { body, ...options });
    }
    /**
     * Returns the specified user profile.
     */
    retrieve(userID, options) {
        return this._client.get(path `/profiles/${userID}`, options);
    }
    /**
     * Update a profile
     */
    update(userID, body, options) {
        return this._client.patch(path `/profiles/${userID}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Deletes the specified user profile.
     */
    delete(userID, options) {
        return this._client.delete(path `/profiles/${userID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * When using `PUT`, be sure to include all the key-value pairs required by the
     * recipient's profile. Any key-value pairs that exist in the profile but fail to
     * be included in the `PUT` request will be removed from the profile. Remember, a
     * `PUT` update is a full replacement of the data. For partial updates, use the
     * [Patch](https://www.courier.com/docs/reference/profiles/patch/) request.
     */
    replace(userID, body, options) {
        return this._client.put(path `/profiles/${userID}`, { body, ...options });
    }
}
Profiles.Lists = Lists;
