// @trycourier/courier/resources/profiles/profiles.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Profiles = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const ListsAPI = tslib_1.__importStar(require("./lists.js"));
const lists_1 = require("./lists.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Profiles extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.lists = new ListsAPI.Lists(this._client);
    }
    /**
     * Merge the supplied values with an existing profile or create a new profile if
     * one doesn't already exist.
     */
    create(userID, body, options) {
        return this._client.post((0, path_1.path) `/profiles/${userID}`, { body, ...options });
    }
    /**
     * Returns the specified user profile.
     */
    retrieve(userID, options) {
        return this._client.get((0, path_1.path) `/profiles/${userID}`, options);
    }
    /**
     * Update a profile
     */
    update(userID, body, options) {
        return this._client.patch((0, path_1.path) `/profiles/${userID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Deletes the specified user profile.
     */
    delete(userID, options) {
        return this._client.delete((0, path_1.path) `/profiles/${userID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * When using `PUT`, be sure to include all the key-value pairs required by the
     * recipient's profile. Any key-value pairs that exist in the profile but fail to
     * be included in the `PUT` request will be removed from the profile. Remember, a
     * `PUT` update is a full replacement of the data. For partial updates, use the
     * [Patch](https://www.courier.com/docs/reference/profiles/patch/) request.
     */
    replace(userID, body, options) {
        return this._client.put((0, path_1.path) `/profiles/${userID}`, { body, ...options });
    }
}
exports.Profiles = Profiles;
Profiles.Lists = lists_1.Lists;
