// @trycourier/courier/resources/users/tenants.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tenants = void 0;
const resource_1 = require("../../core/resource.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Tenants extends resource_1.APIResource {
    /**
     * Returns a paginated list of user tenant associations.
     *
     * @example
     * ```ts
     * const tenants = await client.users.tenants.list('user_id');
     * ```
     */
    list(userID, query = {}, options) {
        return this._client.get((0, path_1.path) `/users/${userID}/tenants`, { query, ...options });
    }
    /**
     * This endpoint is used to add a user to multiple tenants in one call. A custom
     * profile can also be supplied for each tenant. This profile will be merged with
     * the user's main profile when sending to the user with that tenant.
     *
     * @example
     * ```ts
     * await client.users.tenants.addMultiple('user_id', {
     *   tenants: [{ tenant_id: 'tenant_id' }],
     * });
     * ```
     */
    addMultiple(userID, body, options) {
        return this._client.put((0, path_1.path) `/users/${userID}/tenants`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * This endpoint is used to add a single tenant.
     *
     * A custom profile can also be supplied with the tenant. This profile will be
     * merged with the user's main profile when sending to the user with that tenant.
     *
     * @example
     * ```ts
     * await client.users.tenants.addSingle('tenant_id', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    addSingle(tenantID, params, options) {
        const { user_id, ...body } = params;
        return this._client.put((0, path_1.path) `/users/${user_id}/tenants/${tenantID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Removes a user from any tenants they may have been associated with.
     *
     * @example
     * ```ts
     * await client.users.tenants.removeAll('user_id');
     * ```
     */
    removeAll(userID, options) {
        return this._client.delete((0, path_1.path) `/users/${userID}/tenants`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Removes a user from the supplied tenant.
     *
     * @example
     * ```ts
     * await client.users.tenants.removeSingle('tenant_id', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    removeSingle(tenantID, params, options) {
        const { user_id } = params;
        return this._client.delete((0, path_1.path) `/users/${user_id}/tenants/${tenantID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Tenants = Tenants;
