// @trycourier/courier/resources/users/tenants.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Tenants extends APIResource {
    /**
     * Returns a paginated list of user tenant associations.
     *
     * @example
     * ```ts
     * const tenants = await client.users.tenants.list('user_id');
     * ```
     */
    list(userID, query = {}, options) {
        return this._client.get(path `/users/${userID}/tenants`, { query, ...options });
    }
    /**
     * This endpoint is used to add a user to multiple tenants in one call. A custom
     * profile can also be supplied for each tenant. This profile will be merged with
     * the user's main profile when sending to the user with that tenant.
     *
     * @example
     * ```ts
     * await client.users.tenants.addMultiple('user_id', {
     *   tenants: [{ tenant_id: 'tenant_id' }],
     * });
     * ```
     */
    addMultiple(userID, body, options) {
        return this._client.put(path `/users/${userID}/tenants`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * This endpoint is used to add a single tenant.
     *
     * A custom profile can also be supplied with the tenant. This profile will be
     * merged with the user's main profile when sending to the user with that tenant.
     *
     * @example
     * ```ts
     * await client.users.tenants.addSingle('tenant_id', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    addSingle(tenantID, params, options) {
        const { user_id, ...body } = params;
        return this._client.put(path `/users/${user_id}/tenants/${tenantID}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Removes a user from any tenants they may have been associated with.
     *
     * @example
     * ```ts
     * await client.users.tenants.removeAll('user_id');
     * ```
     */
    removeAll(userID, options) {
        return this._client.delete(path `/users/${userID}/tenants`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Removes a user from the supplied tenant.
     *
     * @example
     * ```ts
     * await client.users.tenants.removeSingle('tenant_id', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    removeSingle(tenantID, params, options) {
        const { user_id } = params;
        return this._client.delete(path `/users/${user_id}/tenants/${tenantID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
