// @trycourier/courier/resources/users/tokens.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tokens = void 0;
const resource_1 = require("../../core/resource.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Tokens extends resource_1.APIResource {
    /**
     * Get single token available for a `:token`
     *
     * @example
     * ```ts
     * const token = await client.users.tokens.retrieve('token', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    retrieve(token, params, options) {
        const { user_id } = params;
        return this._client.get((0, path_1.path) `/users/${user_id}/tokens/${token}`, options);
    }
    /**
     * Apply a JSON Patch (RFC 6902) to the specified token.
     *
     * @example
     * ```ts
     * await client.users.tokens.update('token', {
     *   user_id: 'user_id',
     *   patch: [{ op: 'op', path: 'path' }],
     * });
     * ```
     */
    update(token, params, options) {
        const { user_id, ...body } = params;
        return this._client.patch((0, path_1.path) `/users/${user_id}/tokens/${token}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Gets all tokens available for a :user_id
     *
     * @example
     * ```ts
     * const userTokens = await client.users.tokens.list(
     *   'user_id',
     * );
     * ```
     */
    list(userID, options) {
        return this._client.get((0, path_1.path) `/users/${userID}/tokens`, options);
    }
    /**
     * Delete User Token
     *
     * @example
     * ```ts
     * await client.users.tokens.delete('token', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    delete(token, params, options) {
        const { user_id } = params;
        return this._client.delete((0, path_1.path) `/users/${user_id}/tokens/${token}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Adds multiple tokens to a user and overwrites matching existing tokens.
     *
     * @example
     * ```ts
     * await client.users.tokens.addMultiple('user_id');
     * ```
     */
    addMultiple(userID, options) {
        return this._client.put((0, path_1.path) `/users/${userID}/tokens`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Adds a single token to a user and overwrites a matching existing token.
     *
     * @example
     * ```ts
     * await client.users.tokens.addSingle('token', {
     *   user_id: 'user_id',
     *   body_token: 'token',
     *   provider_key: 'firebase-fcm',
     * });
     * ```
     */
    addSingle(pathToken, params, options) {
        const { user_id, ...body } = params;
        return this._client.put((0, path_1.path) `/users/${user_id}/tokens/${pathToken}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Tokens = Tokens;
