// @trycourier/courier/resources/users/tokens.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Tokens extends APIResource {
    /**
     * Get single token available for a `:token`
     *
     * @example
     * ```ts
     * const token = await client.users.tokens.retrieve('token', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    retrieve(token, params, options) {
        const { user_id } = params;
        return this._client.get(path `/users/${user_id}/tokens/${token}`, options);
    }
    /**
     * Apply a JSON Patch (RFC 6902) to the specified token.
     *
     * @example
     * ```ts
     * await client.users.tokens.update('token', {
     *   user_id: 'user_id',
     *   patch: [{ op: 'op', path: 'path' }],
     * });
     * ```
     */
    update(token, params, options) {
        const { user_id, ...body } = params;
        return this._client.patch(path `/users/${user_id}/tokens/${token}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Gets all tokens available for a :user_id
     *
     * @example
     * ```ts
     * const userTokens = await client.users.tokens.list(
     *   'user_id',
     * );
     * ```
     */
    list(userID, options) {
        return this._client.get(path `/users/${userID}/tokens`, options);
    }
    /**
     * Delete User Token
     *
     * @example
     * ```ts
     * await client.users.tokens.delete('token', {
     *   user_id: 'user_id',
     * });
     * ```
     */
    delete(token, params, options) {
        const { user_id } = params;
        return this._client.delete(path `/users/${user_id}/tokens/${token}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Adds multiple tokens to a user and overwrites matching existing tokens.
     *
     * @example
     * ```ts
     * await client.users.tokens.addMultiple('user_id');
     * ```
     */
    addMultiple(userID, options) {
        return this._client.put(path `/users/${userID}/tokens`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Adds a single token to a user and overwrites a matching existing token.
     *
     * @example
     * ```ts
     * await client.users.tokens.addSingle('token', {
     *   user_id: 'user_id',
     *   body_token: 'token',
     *   provider_key: 'firebase-fcm',
     * });
     * ```
     */
    addSingle(pathToken, params, options) {
        const { user_id, ...body } = params;
        return this._client.put(path `/users/${user_id}/tokens/${pathToken}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
