// @trycourier/courier/resources/tenants/tenants.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as TemplatesAPI from "./templates.mjs";
import { Templates } from "./templates.mjs";
import * as PreferencesAPI from "./preferences/preferences.mjs";
import { Preferences } from "./preferences/preferences.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Tenants extends APIResource {
    constructor() {
        super(...arguments);
        this.preferences = new PreferencesAPI.Preferences(this._client);
        this.templates = new TemplatesAPI.Templates(this._client);
    }
    /**
     * Get a Tenant
     *
     * @example
     * ```ts
     * const tenant = await client.tenants.retrieve('tenant_id');
     * ```
     */
    retrieve(tenantID, options) {
        return this._client.get(path `/tenants/${tenantID}`, options);
    }
    /**
     * Create or Replace a Tenant
     *
     * @example
     * ```ts
     * const tenant = await client.tenants.update('tenant_id', {
     *   name: 'name',
     * });
     * ```
     */
    update(tenantID, body, options) {
        return this._client.put(path `/tenants/${tenantID}`, { body, ...options });
    }
    /**
     * Get a List of Tenants
     *
     * @example
     * ```ts
     * const tenants = await client.tenants.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/tenants', { query, ...options });
    }
    /**
     * Delete a Tenant
     *
     * @example
     * ```ts
     * await client.tenants.delete('tenant_id');
     * ```
     */
    delete(tenantID, options) {
        return this._client.delete(path `/tenants/${tenantID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get Users in Tenant
     *
     * @example
     * ```ts
     * const response = await client.tenants.listUsers(
     *   'tenant_id',
     * );
     * ```
     */
    listUsers(tenantID, query = {}, options) {
        return this._client.get(path `/tenants/${tenantID}/users`, { query, ...options });
    }
}
Tenants.Preferences = Preferences;
Tenants.Templates = Templates;
