// @trycourier/courier/resources/tenants/tenants.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Tenants = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const TemplatesAPI = tslib_1.__importStar(require("./templates.js"));
const templates_1 = require("./templates.js");
const PreferencesAPI = tslib_1.__importStar(require("./preferences/preferences.js"));
const preferences_1 = require("./preferences/preferences.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Tenants extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.preferences = new PreferencesAPI.Preferences(this._client);
        this.templates = new TemplatesAPI.Templates(this._client);
    }
    /**
     * Get a Tenant
     *
     * @example
     * ```ts
     * const tenant = await client.tenants.retrieve('tenant_id');
     * ```
     */
    retrieve(tenantID, options) {
        return this._client.get((0, path_1.path) `/tenants/${tenantID}`, options);
    }
    /**
     * Create or Replace a Tenant
     *
     * @example
     * ```ts
     * const tenant = await client.tenants.update('tenant_id', {
     *   name: 'name',
     * });
     * ```
     */
    update(tenantID, body, options) {
        return this._client.put((0, path_1.path) `/tenants/${tenantID}`, { body, ...options });
    }
    /**
     * Get a List of Tenants
     *
     * @example
     * ```ts
     * const tenants = await client.tenants.list();
     * ```
     */
    list(query = {}, options) {
        return this._client.get('/tenants', { query, ...options });
    }
    /**
     * Delete a Tenant
     *
     * @example
     * ```ts
     * await client.tenants.delete('tenant_id');
     * ```
     */
    delete(tenantID, options) {
        return this._client.delete((0, path_1.path) `/tenants/${tenantID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get Users in Tenant
     *
     * @example
     * ```ts
     * const response = await client.tenants.listUsers(
     *   'tenant_id',
     * );
     * ```
     */
    listUsers(tenantID, query = {}, options) {
        return this._client.get((0, path_1.path) `/tenants/${tenantID}/users`, { query, ...options });
    }
}
exports.Tenants = Tenants;
Tenants.Preferences = preferences_1.Preferences;
Tenants.Templates = templates_1.Templates;
