// @trycourier/courier/resources/lists/lists.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as SubscriptionsAPI from "./subscriptions.mjs";
import { Subscriptions, } from "./subscriptions.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Lists extends APIResource {
    constructor() {
        super(...arguments);
        this.subscriptions = new SubscriptionsAPI.Subscriptions(this._client);
    }
    /**
     * Returns a list based on the list ID provided.
     */
    retrieve(listID, options) {
        return this._client.get(path `/lists/${listID}`, options);
    }
    /**
     * Create or replace an existing list with the supplied values.
     */
    update(listID, body, options) {
        return this._client.put(path `/lists/${listID}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Returns all of the lists, with the ability to filter based on a pattern.
     */
    list(query = {}, options) {
        return this._client.get('/lists', { query, ...options });
    }
    /**
     * Delete a list by list ID.
     */
    delete(listID, options) {
        return this._client.delete(path `/lists/${listID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Restore a previously deleted list.
     */
    restore(listID, body, options) {
        return this._client.put(path `/lists/${listID}/restore`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
Lists.Subscriptions = Subscriptions;
