// @trycourier/courier/resources/lists/lists.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Lists = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const SubscriptionsAPI = tslib_1.__importStar(require("./subscriptions.js"));
const subscriptions_1 = require("./subscriptions.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Lists extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.subscriptions = new SubscriptionsAPI.Subscriptions(this._client);
    }
    /**
     * Returns a list based on the list ID provided.
     */
    retrieve(listID, options) {
        return this._client.get((0, path_1.path) `/lists/${listID}`, options);
    }
    /**
     * Create or replace an existing list with the supplied values.
     */
    update(listID, body, options) {
        return this._client.put((0, path_1.path) `/lists/${listID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Returns all of the lists, with the ability to filter based on a pattern.
     */
    list(query = {}, options) {
        return this._client.get('/lists', { query, ...options });
    }
    /**
     * Delete a list by list ID.
     */
    delete(listID, options) {
        return this._client.delete((0, path_1.path) `/lists/${listID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Restore a previously deleted list.
     */
    restore(listID, body, options) {
        return this._client.put((0, path_1.path) `/lists/${listID}/restore`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Lists = Lists;
Lists.Subscriptions = subscriptions_1.Subscriptions;
