// @trycourier/courier/resources/lists/subscriptions.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscriptions = void 0;
const resource_1 = require("../../core/resource.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Subscriptions extends resource_1.APIResource {
    /**
     * Get the list's subscriptions.
     */
    list(listID, query = {}, options) {
        return this._client.get((0, path_1.path) `/lists/${listID}/subscriptions`, { query, ...options });
    }
    /**
     * Subscribes additional users to the list, without modifying existing
     * subscriptions. If the list does not exist, it will be automatically created.
     */
    add(listID, body, options) {
        return this._client.post((0, path_1.path) `/lists/${listID}/subscriptions`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Subscribes the users to the list, overwriting existing subscriptions. If the
     * list does not exist, it will be automatically created.
     */
    subscribe(listID, body, options) {
        return this._client.put((0, path_1.path) `/lists/${listID}/subscriptions`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Subscribe a user to an existing list (note: if the List does not exist, it will
     * be automatically created).
     */
    subscribeUser(userID, params, options) {
        const { list_id, ...body } = params;
        return this._client.put((0, path_1.path) `/lists/${list_id}/subscriptions/${userID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Delete a subscription to a list by list ID and user ID.
     */
    unsubscribeUser(userID, params, options) {
        const { list_id } = params;
        return this._client.delete((0, path_1.path) `/lists/${list_id}/subscriptions/${userID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Subscriptions = Subscriptions;
