// @trycourier/courier/resources/lists/subscriptions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
export class Subscriptions extends APIResource {
    /**
     * Get the list's subscriptions.
     */
    list(listID, query = {}, options) {
        return this._client.get(path `/lists/${listID}/subscriptions`, { query, ...options });
    }
    /**
     * Subscribes additional users to the list, without modifying existing
     * subscriptions. If the list does not exist, it will be automatically created.
     */
    add(listID, body, options) {
        return this._client.post(path `/lists/${listID}/subscriptions`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Subscribes the users to the list, overwriting existing subscriptions. If the
     * list does not exist, it will be automatically created.
     */
    subscribe(listID, body, options) {
        return this._client.put(path `/lists/${listID}/subscriptions`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Subscribe a user to an existing list (note: if the List does not exist, it will
     * be automatically created).
     */
    subscribeUser(userID, params, options) {
        const { list_id, ...body } = params;
        return this._client.put(path `/lists/${list_id}/subscriptions/${userID}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Delete a subscription to a list by list ID and user ID.
     */
    unsubscribeUser(userID, params, options) {
        const { list_id } = params;
        return this._client.delete(path `/lists/${list_id}/subscriptions/${userID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
