// @trycourier/courier/resources/audiences.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Audiences = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Audiences extends resource_1.APIResource {
    /**
     * Returns the specified audience by id.
     */
    retrieve(audienceID, options) {
        return this._client.get((0, path_1.path) `/audiences/${audienceID}`, options);
    }
    /**
     * Creates or updates audience.
     */
    update(audienceID, body, options) {
        return this._client.put((0, path_1.path) `/audiences/${audienceID}`, { body, ...options });
    }
    /**
     * Get the audiences associated with the authorization token.
     */
    list(query = {}, options) {
        return this._client.get('/audiences', { query, ...options });
    }
    /**
     * Deletes the specified audience.
     */
    delete(audienceID, options) {
        return this._client.delete((0, path_1.path) `/audiences/${audienceID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get list of members of an audience.
     */
    listMembers(audienceID, query = {}, options) {
        return this._client.get((0, path_1.path) `/audiences/${audienceID}/members`, { query, ...options });
    }
}
exports.Audiences = Audiences;
