// @trycourier/courier/resources/audiences.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class Audiences extends APIResource {
    /**
     * Returns the specified audience by id.
     */
    retrieve(audienceID, options) {
        return this._client.get(path `/audiences/${audienceID}`, options);
    }
    /**
     * Creates or updates audience.
     */
    update(audienceID, body, options) {
        return this._client.put(path `/audiences/${audienceID}`, { body, ...options });
    }
    /**
     * Get the audiences associated with the authorization token.
     */
    list(query = {}, options) {
        return this._client.get('/audiences', { query, ...options });
    }
    /**
     * Deletes the specified audience.
     */
    delete(audienceID, options) {
        return this._client.delete(path `/audiences/${audienceID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Get list of members of an audience.
     */
    listMembers(audienceID, query = {}, options) {
        return this._client.get(path `/audiences/${audienceID}/members`, { query, ...options });
    }
}
