// @trycourier/courier/resources/brands.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class Brands extends APIResource {
    /**
     * Create a new brand
     */
    create(body, options) {
        return this._client.post('/brands', { body, ...options });
    }
    /**
     * Fetch a specific brand by brand ID.
     */
    retrieve(brandID, options) {
        return this._client.get(path `/brands/${brandID}`, options);
    }
    /**
     * Replace an existing brand with the supplied values.
     */
    update(brandID, body, options) {
        return this._client.put(path `/brands/${brandID}`, { body, ...options });
    }
    /**
     * Get the list of brands.
     */
    list(query = {}, options) {
        return this._client.get('/brands', { query, ...options });
    }
    /**
     * Delete a brand by brand ID.
     */
    delete(brandID, options) {
        return this._client.delete(path `/brands/${brandID}`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
