// @trycourier/courier/resources/brands.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Brands = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Brands extends resource_1.APIResource {
    /**
     * Create a new brand
     */
    create(body, options) {
        return this._client.post('/brands', { body, ...options });
    }
    /**
     * Fetch a specific brand by brand ID.
     */
    retrieve(brandID, options) {
        return this._client.get((0, path_1.path) `/brands/${brandID}`, options);
    }
    /**
     * Replace an existing brand with the supplied values.
     */
    update(brandID, body, options) {
        return this._client.put((0, path_1.path) `/brands/${brandID}`, { body, ...options });
    }
    /**
     * Get the list of brands.
     */
    list(query = {}, options) {
        return this._client.get('/brands', { query, ...options });
    }
    /**
     * Delete a brand by brand ID.
     */
    delete(brandID, options) {
        return this._client.delete((0, path_1.path) `/brands/${brandID}`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Brands = Brands;
