// @trycourier/courier/resources/bulk.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class Bulk extends APIResource {
    /**
     * Ingest user data into a Bulk Job
     */
    addUsers(jobID, body, options) {
        return this._client.post(path `/bulk/${jobID}`, {
            body,
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Create a bulk job
     */
    createJob(body, options) {
        return this._client.post('/bulk', { body, ...options });
    }
    /**
     * Get Bulk Job Users
     */
    listUsers(jobID, query = {}, options) {
        return this._client.get(path `/bulk/${jobID}/users`, { query, ...options });
    }
    /**
     * Get a bulk job
     */
    retrieveJob(jobID, options) {
        return this._client.get(path `/bulk/${jobID}`, options);
    }
    /**
     * Run a bulk job
     */
    runJob(jobID, options) {
        return this._client.post(path `/bulk/${jobID}/run`, {
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
