// @trycourier/courier/resources/bulk.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Bulk = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
class Bulk extends resource_1.APIResource {
    /**
     * Ingest user data into a Bulk Job
     */
    addUsers(jobID, body, options) {
        return this._client.post((0, path_1.path) `/bulk/${jobID}`, {
            body,
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Create a bulk job
     */
    createJob(body, options) {
        return this._client.post('/bulk', { body, ...options });
    }
    /**
     * Get Bulk Job Users
     */
    listUsers(jobID, query = {}, options) {
        return this._client.get((0, path_1.path) `/bulk/${jobID}/users`, { query, ...options });
    }
    /**
     * Get a bulk job
     */
    retrieveJob(jobID, options) {
        return this._client.get((0, path_1.path) `/bulk/${jobID}`, options);
    }
    /**
     * Run a bulk job
     */
    runJob(jobID, options) {
        return this._client.post((0, path_1.path) `/bulk/${jobID}/run`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.Bulk = Bulk;
