// @trycourier/courier/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Users = exports.Translations = exports.Tenants = exports.Send = exports.Requests = exports.Profiles = exports.Notifications = exports.Messages = exports.Lists = exports.Inbound = exports.Bulk = exports.Brands = exports.Automations = exports.Auth = exports.AuditEvents = exports.Audiences = void 0;
const tslib_1 = require("../internal/tslib.js");
tslib_1.__exportStar(require("./shared.js"), exports);
var audiences_1 = require("./audiences.js");
Object.defineProperty(exports, "Audiences", { enumerable: true, get: function () { return audiences_1.Audiences; } });
var audit_events_1 = require("./audit-events.js");
Object.defineProperty(exports, "AuditEvents", { enumerable: true, get: function () { return audit_events_1.AuditEvents; } });
var auth_1 = require("./auth.js");
Object.defineProperty(exports, "Auth", { enumerable: true, get: function () { return auth_1.Auth; } });
var automations_1 = require("./automations/automations.js");
Object.defineProperty(exports, "Automations", { enumerable: true, get: function () { return automations_1.Automations; } });
var brands_1 = require("./brands.js");
Object.defineProperty(exports, "Brands", { enumerable: true, get: function () { return brands_1.Brands; } });
var bulk_1 = require("./bulk.js");
Object.defineProperty(exports, "Bulk", { enumerable: true, get: function () { return bulk_1.Bulk; } });
var inbound_1 = require("./inbound.js");
Object.defineProperty(exports, "Inbound", { enumerable: true, get: function () { return inbound_1.Inbound; } });
var lists_1 = require("./lists/lists.js");
Object.defineProperty(exports, "Lists", { enumerable: true, get: function () { return lists_1.Lists; } });
var messages_1 = require("./messages.js");
Object.defineProperty(exports, "Messages", { enumerable: true, get: function () { return messages_1.Messages; } });
var notifications_1 = require("./notifications/notifications.js");
Object.defineProperty(exports, "Notifications", { enumerable: true, get: function () { return notifications_1.Notifications; } });
var profiles_1 = require("./profiles/profiles.js");
Object.defineProperty(exports, "Profiles", { enumerable: true, get: function () { return profiles_1.Profiles; } });
var requests_1 = require("./requests.js");
Object.defineProperty(exports, "Requests", { enumerable: true, get: function () { return requests_1.Requests; } });
var send_1 = require("./send.js");
Object.defineProperty(exports, "Send", { enumerable: true, get: function () { return send_1.Send; } });
var tenants_1 = require("./tenants/tenants.js");
Object.defineProperty(exports, "Tenants", { enumerable: true, get: function () { return tenants_1.Tenants; } });
var translations_1 = require("./translations.js");
Object.defineProperty(exports, "Translations", { enumerable: true, get: function () { return translations_1.Translations; } });
var users_1 = require("./users/users.js");
Object.defineProperty(exports, "Users", { enumerable: true, get: function () { return users_1.Users; } });
