// @trycourier/courier/resources/messages.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
export class Messages extends APIResource {
    /**
     * Fetch the status of a message you've previously sent.
     */
    retrieve(messageID, options) {
        return this._client.get(path `/messages/${messageID}`, options);
    }
    /**
     * Fetch the statuses of messages you've previously sent.
     */
    list(query = {}, options) {
        return this._client.get('/messages', { query, ...options });
    }
    /**
     * Cancel a message that is currently in the process of being delivered. A
     * well-formatted API call to the cancel message API will return either `200`
     * status code for a successful cancellation or `409` status code for an
     * unsuccessful cancellation. Both cases will include the actual message record in
     * the response body (see details below).
     */
    cancel(messageID, options) {
        return this._client.post(path `/messages/${messageID}/cancel`, options);
    }
    /**
     * Get message content
     */
    content(messageID, options) {
        return this._client.get(path `/messages/${messageID}/output`, options);
    }
    /**
     * Fetch the array of events of a message you've previously sent.
     */
    history(messageID, query = {}, options) {
        return this._client.get(path `/messages/${messageID}/history`, { query, ...options });
    }
}
